src/ComServerHelpers/GeneralClassFactory.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.Versioning;
+
+namespace ComServerHelpers;
+
+/// <summary>
+/// General class factory for .NET types using generics.
+/// </summary>
+/// <typeparam name="T">Type the factory creates.</typeparam>
+/// <typeparam name="TInterface">Interface that <typeparamref name="T"/> implements.</typeparam>
+/// <seealso cref="BaseClassFactory"/>
+[SupportedOSPlatform("windows6.0.6000")]
+public sealed class GeneralClassFactory<T, TInterface> : BaseClassFactory where T : class, TInterface, new()
+{
+    /// <inheritdoc/>
+    protected internal override Guid Clsid => typeof(T).GUID;
```

Comment:
This is not really AOT-friendly, you should avoid doing `Type.GUID`.

Reply:
Heads up that `typeof(T).GUID` can return different results on NAOT vs CoreCLR for types without the `GuidAttribute` applied. This is especially important for any usage of built-in COM.

This is why we have the `IIUnknownInterfaceDetails` abstraction for source-generated COM.

Reply:
I would note that this is just one implementation of `BaseClassFactory`. If we want an AOT-friendly factory, there's no reason we can't make one that uses Source Generators to read from `[Guid]`

Reply:
I'm going to leave it as is for the first iteration and make it `a requirement` to annotate the server class with `Guid` attribute.
I agree with Shmuelie that source generator is a better approach and I will adopt it in the next iteration.